### PR TITLE
Handle metricscollector case worker container have no command

### PR DESCRIPTION
**Pull request by hougangliu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/914)

### CODE REVIEW

File: pkg/webhook/v1alpha3/pod/utils.go [hunk 1]
```diff
@@ -74,3 +80,64 @@ func getLabel(pod *v1.Pod, targetLabel string) (string, error) {
...
+		Namespace:          namespace,
+		ServiceAccountName: pod.Spec.ServiceAccountName,
+		ImagePullSecrets:   imagePullSecrets,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create k8schain: %v", err)
+	}
+
+	mkc := authn.NewMultiKeychain(kc)
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(mkc))
```
Comment (gaocegege) [inline pkg/webhook/v1alpha3/pod/utils.go]: What will be created when we call it? Will we get the image from the remote registry in a blocking manner?

Comment (hougangliu) [reply pkg/webhook/v1alpha3/pod/utils.go]: it only fetches manifest instead of image file

File: Gopkg.toml [hunk 2]
```diff
@@ -48,28 +53,32 @@ required = [
...
+
+[[override]]
+  name = "k8s.io/apiextensions-apiserver"
+  version = "kubernetes-1.12.9"
 
 [[override]]
-  name="k8s.io/apiextensions-apiserver"
-  version="kubernetes-1.12.3"
+  name = "k8s.io/kubernetes"
+  version = "v1.13.3"
```
Comment (gaocegege) [inline Gopkg.toml]: Not sure if we can upgrade now. /cc @johnugeorge

File: pkg/webhook/v1alpha3/pod/utils.go [hunk 3]
```diff
@@ -74,3 +80,64 @@ func getLabel(pod *v1.Pod, targetLabel string) (string, error) {
...
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get container image %q info from registry: %v", c.Image, err)
+	}
+
+	return img, nil
+}
+
+func getImageCommand(pod *v1.Pod, namespace string, containerIndex int) ([]string, error) {
```
Comment (gaocegege) [inline pkg/webhook/v1alpha3/pod/utils.go]: The name is confusing. We may not get the command from the image if the container has command. How about getContainerCommand?

File: Gopkg.toml (hunk 2 above)
Comment (hougangliu) [reply Gopkg.toml]: I tried, it works well.

File: pkg/webhook/v1alpha3/pod/utils.go (hunk 3 above)
Comment (hougangliu) [reply pkg/webhook/v1alpha3/pod/utils.go]: agree with you

**Comment by gaocegege** [comment]
/lgtm
/approve

@johnugeorge If you have problems, we can revert it. Prefer not to block @hougangliu 's work.

**Pull request merged**